qiling/os/posix/syscall/unistd.py

```diff
@@ -338,23 +338,25 @@ def ql_syscall_write(ql: Qiling, fd: int, buf: int, count: int):
 def ql_syscall_readlink(ql: Qiling, path_name: int, path_buff: int, path_buffsize: int):
     pathname = ql.os.utils.read_cstring(path_name)
     # pathname = str(pathname, 'utf-8', errors="ignore")
-    real_path = ql.os.path.transform_to_link_path(pathname)
-    relative_path = ql.os.path.transform_to_relative_path(pathname)
+    host_path = ql.os.path.virtual_to_host_path(pathname)
+    virt_path = ql.os.path.virtual_abspath(pathname)
 
-    if not os.path.exists(real_path):
-        regreturn = -1
+    # cover procfs psaudo files first
+    # TODO: /proc/self/root, /proc/self/cwd
+    if virt_path == r'/proc/self/exe':
```

Comment:
Maybe a separate file for such things instead of “hacky if”?

Reply:
I agree this is ~hacky~ ugly, but that `if` was already there..
The elegant way to handle this would be a fully virtualized file system that could handle symlinks like this one.